green-hub/src/components/Ideas/Ideas.jsx

```diff
@@ -1,116 +1,95 @@
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import { Navbar, Footer } from "../index";
 import { db } from "../../config/firebase.js";
 import {
-  Timestamp,
-  addDoc,
-  // getDocs,
-  collection,
-  orderBy,
-  onSnapshot,
-  query,
-  getDoc,
-  doc
+	Timestamp,
+	addDoc,
+	collection,
+	orderBy,
+	onSnapshot,
+	query,
+	getDoc,
+	doc,
 } from "firebase/firestore";
 import { Toaster, toast } from "react-hot-toast";
 import IdeaCard from "../IdeaCard/IdeaCard";
 import styles from "./Ideas.module.css";
 import useAuth from "../../context/AuthContext";
 
 const Ideas = () => {
-  const [idea, setIdea] = useState([]);
-  const [user, setUser] = useState('')
-  const ideasCollectionRef = collection(db, "ideas");
-  // const usersCollectionRef = collection(db, "users")
-  const { currentUser } = useAuth();
 
-  // POBIERANIE USERÓW
-  // const getUser = async () => {
-  //   try {
-  //     const data = await getDocs(usersCollectionRef);
-  //     const filteredData = data.docs.map((doc) => ({
-  //       ...doc.data()
-  //     }))
-  //     console.log('fileteredData: ', filteredData)
-  //     const userData = filteredData?.filter(el => el.email === currentUser.email)
-  //     console.log('user: ', userData, 'currentUser: ', currentUser)
-  //     const userName = `${userData[0].name} ${userData[0].lastName}`
-  //     console.log("user - czy jesteś tu: ", userName)
-  //     setUser(userName);
-  //   } catch (error) {
-  //     console.log("no user here", currentUser);
-  //     console.error(error)
-  //   }
-  // }
+	const [idea, setIdea] = useState([]);
+	const [user, setUser] = useState("");
+	const ideasCollectionRef = collection(db, "ideas");
+	const { currentUser } = useAuth();
 
+	// POBIERANIE USERÓW
+		const getUserName = async () => {
+		try {
+			const userData = await getDoc(doc(db, "users", currentUser?.uid));
```

Comment:
podejrzewam, że pobieracie te dane już gdzieś wyżej w aplikacji - może tutaj już nie trzeba tego robić znowu?

Reply:
robimy to w dwóch miejscach, ale tutaj pobieramy imię i nazwisko, a w Home pobieramy tylko imię dlatego nie wyrzuciłam tej funkcji do oddzielnego pliku...